browser/resources/settings/brave_settings_overrides.js

```diff
@@ -395,6 +395,22 @@ BravePatching.RegisterPolymerTemplateModifications({
       version.innerHTML = '<a id="release-notes" target="_blank" href="https://brave.com/latest/">' + version.innerHTML + '</a>'
     }
   },
+  'settings-clear-browsing-data-dialog': (templateContent) => {
+    const advancedTab = templateContent.querySelector('#advanced-tab')
+    const shieldsSettingHTML = `
+      <settings-checkbox
+        pref="{{prefs.browser.clear_data.shields_settings}}"
+        label="${I18nBehavior.i18n('clearShieldsSettings')}"
+        sub-label="[[counters_.shields_settings]]"
+        disabled="[[clearingInProgress_]]">
+      </settings-checkbox>
+    `
+
+    let tempParent = document.createElement('div')
+    tempParent.innerHTML = shieldsSettingHTML.trim()
+    // Insert after Cached images and files checkbox. It's fourth child.
+    advancedTab.children[4].insertAdjacentElement('afterend', tempParent.firstChild)
```

Comment:
Nit: Maybe check to see if the children length is > 4 and both throw an error and insert at the end if it's not. So we can catch this during rebase.

Reply:
Done.